### PR TITLE
fix(chainMapping): fix typos in chain mappings

### PR DESCRIPTION
**Pull request by roninjin10**
Copy paste pasta broke chain mappings.  This is causing optimism bridge to fail because l1 contracts get loaded instead of l2

image

### CODE REVIEW

**Comment by roninjin10** [comment]
Ah yes you are right.   Going to step through the code live now to debug really quick.

**Comment by roninjin10** [comment]
So when I step in I'm in this minimized code.   Will look at the object next
image

**Comment by roninjin10** [comment]
I see the issue now.  It expects an object ({id: 10}) not an id (10).  I think this is pretty confusing and got a way of being a bit safer here

File: index.js [hunk 1]
```diff
@@ -51,7 +49,12 @@ const chainIdMapping = Object.entries({
 	return memo;
 }, {});
 
-const getNetworkFromId = ({ id }) => chainIdMapping[id];
+/** @type {(obj: {id: number} | number) => number} */
+const getNetworkFromId = (obj) => {
+	const id = typeof obj === 'number' ? obj : obj.id;
```
Comment (roninjin10) [inline index.js:54]: The name is pretty confusing and it fails silently if you pass in a number id (which based on the name makes sense).   The two solutions are this where I handle a number being passed in (without making a breaking change).   OR throwing an error (which is kinda a breaking change if people relied on it not throwing an error before).

**Comment by roninjin10** [comment]
@bachstatter I updated pr based on more throrough debugging.  Let me know what you think

File: index.js [hunk 2]
```diff
@@ -31,10 +31,8 @@ const chainIdMapping = Object.entries({
 		fork: true,
 	},
 
-	// OVM networks: see https://github.com/ethereum-optimism/regenesis/
 	10: {
 		network: 'mainnet',
```
Comment (roninjin10) [inline index.js:35]:
```suggestion
		network: 'mainnet',
		useOvm: true,
```

File: index.js (hunk 1 above)
Comment (bachstatter) [reply index.js:55]: Yea this seems fine to me

File: index.js [hunk 3]
```diff
@@ -51,7 +50,12 @@ const chainIdMapping = Object.entries({
 	return memo;
 }, {});
 
-const getNetworkFromId = ({ id }) => chainIdMapping[id];
+/** @type {(obj: {id: number} | number) => number} */
+const getNetworkFromId = (obj) => {
+	const id = typeof obj === 'number' ? obj : obj.id;
+	chainIdMapping[id];
```
Comment (bachstatter) [inline index.js:56]: Dont forget to return this please :)

Comment (roninjin10) [reply index.js:56]: lol I'm incapable of writing a single line of working code without typescript it appears
```suggestion
	return chainIdMapping[id];
```

**Review by bachstatter — APPROVED** [review]

**Pull request merged**